csds-client/client/client.go

```diff
@@ -0,0 +1,188 @@
+package client
+
+import (
+	"context"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	csdspb "github.com/envoyproxy/go-control-plane/envoy/service/status/v2"
+	envoy_type_matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+	"google.golang.org/grpc/metadata"
+)
+
+type Flag struct {
+	uri         string
+	platform    string
+	authnMode   string
+	apiVersion  string
+	RequestFile string
+	RequestYaml string
+	jwt         string
+	configFile  string
+}
+
+type Client struct {
+	Cc         *grpc.ClientConn
+	CsdsClient csdspb.ClientStatusDiscoveryServiceClient
+
+	Nm   []*envoy_type_matcher.NodeMatcher
+	Md   metadata.MD
+	Info Flag
+}
+
+// parse flags to info
+func ParseFlags() Flag {
+	uriPtr := flag.String("service_uri", "trafficdirector.googleapis.com:443", "the uri of the service to connect to")
+	platformPtr := flag.String("cloud_platform", "gcp", "the cloud platform (e.g. gcp, aws,  ...)")
+	authnModePtr := flag.String("authn_mode", "auto", "the method to use for authentication (e.g. auto, jwt, ...)")
+	apiVersionPtr := flag.String("api_version", "v2", "which xds api major version  to use (e.g. v2, v3 ...)")
+	requestFilePtr := flag.String("request_file", "", "yaml file that defines the csds request")
+	requestYamlPtr := flag.String("request_yaml", "", "yaml string that defines the csds request")
+	jwtPtr := flag.String("jwt_file", "", "path of the -jwt_file")
+	configFilePtr := flag.String("file_to_save_config", "", "the file name to save config")
+
+	flag.Parse()
+
+	f := Flag{
+		uri:         *uriPtr,
+		platform:    *platformPtr,
+		authnMode:   *authnModePtr,
+		apiVersion:  *apiVersionPtr,
+		RequestFile: *requestFilePtr,
+		RequestYaml: *requestYamlPtr,
+		jwt:         *jwtPtr,
+		configFile:  *configFilePtr,
+	}
+
+	return f
+}
+
+// parse the csds request yaml to nodematcher
+func (c *Client) ParseNodeMatcher() error {
```

Comment:
Is this an internal for this client function? Does it have to be exported? The same question for all other functions that maybe do not need to be exported.

Reply:
Actually I exported it just for test. Since it calls `ParseYaml()` in util, maybe I can make this function private and only test `ParseYaml()`

Reply:
If you place tests in the client package - you won't need to export.

Reply:
Moved the test to the client package and made the function private.